src/CacheFactory.php

```diff
@@ -48,7 +48,7 @@ protected function createWithConfig(ContainerInterface $container, $configKey)
 
         switch ($config['class']) {
             case FilesystemCache::class:
-                $cache = new $config['class']();
+                $cache = new $config['class']($config['directory']);
```

Comment:
Should an assertion be added here to ensure `directory` key exists in `$config`?